Source/WebCore/rendering/svg/RenderSVGRect.cpp

```diff
@@ -128,6 +122,39 @@ void RenderSVGRect::fillShape(GraphicsContext& context) const
     context.fillRect(m_fillBoundingBox);
 }
 
+bool RenderSVGRect::canUseStrokeHitTestFastPath() const
+{
+    // Non-scaling-stroke needs special handling.
+    if (hasNonScalingStroke())
+        return false;
+
+    // We can compute intersections with simple, continuous strokes on
+    // regular rectangles without using a Path.
+    return m_shapeType == ShapeType::Rectangle && definitelyHasSimpleStroke();
+}
+
+// Returns true if the stroke is continuous and definitely uses miter joins.
```

Comment:
TIL stroke-miterlimit actually applies to `rect` and not just `path`, `line`, and `polygon`!

Reply:
:D